modules/sitemaps.php

```diff
@@ -23,8 +37,11 @@ function jetpack_sitemap_generate_on_activate() {
 	require_once dirname( __FILE__ ) . '/sitemaps/sitemap-builder.php';
 
 	wp_clear_scheduled_hook( 'jp_sitemap_cron_hook' );
-	// Tell build that it's true we're activating this module.
+	wp_clear_scheduled_hook( 'jetpack_sitemap_generate_on_activate' );
+
+	// Tell builder that it's true we're activating this module.
 	$sitemap_builder = new Jetpack_Sitemap_Builder( true );
```

Comment:
Since we're always running sitemap generation asynchronously, do we even need a constructor parameter here? Otherwise everything's great!